src/App.js

```diff
@@ -1,147 +1,116 @@
+          }
+
+          userCity.currentBuildings.forEach(currentBuilding => {
+            setBuildings(currentBuilding.buildingId, currentBuilding.level);
+          });
+        });
+
+        setLoading('loadUserCitys', true);
+      } catch (error) {
+        console.log('loading Game datas Fail');
+      }
     }
-  }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (loadingStatus.loadUserCitys) {
+      setLoading('status', false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loadingStatus]);
 
   useEffect(() => {
-    if (currentBuildingBuild !== null) {
-      const interval = setInterval(
-        () =>
-          setCurrentBuildingBuild(current => {
-            const timeNow = Date.now();
-            const endBuildTime = current.endTime;
-
-            const buildingDiffTime = endBuildTime - timeNow;
-
-            if (buildingDiffTime < 0) {
-              setCurrentBuildings(currentBuilding =>
-                currentBuilding.map(objBuilding => {
-                  if (objBuilding.buildingId === current.id) {
-                    if (current.toLevel !== objBuilding.level) {
-                      const updateToLevel = objBuilding.level + 1;
-
-                      setCurrentResources(current =>
-                        current.map(obj => {
-                          if (obj.id === 5) {
-                            return {...obj, value: productionResources(obj.name, updateToLevel)};
-                          }
-                          return obj;
-                        })
-                      );
-
-                      return {...objBuilding, level: updateToLevel};
-                    }
-                  }
-                  return objBuilding;
-                })
-              );
-            } else {
-              return {...current, diffTime: buildingDiffTime};
-            }
-
-            return null;
-          }),
-        1000
-      );
+    if (!loadingStatus.status) {
+      const interval = setInterval(() => {
+        currentUserBuildings.forEach(building => {
+          if (building.resourcesType !== 'energy') {
+            addResources(building.resourcesType, productionResources(building.resourcesType, building.level));
+          }
+        });
+      }, 1000);
 
       return () => {
         clearInterval(interval);
       };
     }
-  }, [currentBuildingBuild]);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentUserResources, loadingStatus]);
 
   useEffect(() => {
-    const interval = setInterval(
-      () =>
-        setCurrentResources(current =>
-          current.map(obj => {
-            return {...obj, value: obj.value + productionResources(obj.name, 0)};
-          })
-        ),
-      1000
+    if (currentUserBuildingInProgress.length > 0) {
+      const timeNow = Date.now();
+      const endBuildTime = currentUserBuildingInProgress[0].endTime;
```

Comment:
Same here with the first element of an array. You always refer to the first one. I would recommend to change your data structure.

Reply:
yes that is planned here also in such a way that it accesses only first the first element. since there will be still several cities